src/sdk.ts

```diff
@@ -29,40 +27,44 @@ export function init<T>(
     enableCaptureFailedRequests: false,
     ...passedOptions,
   };
-  if (finalOptions.enabled === false ||
-    NATIVE.platform === 'web') {
+  if (finalOptions.enabled === false || NATIVE.platform === 'web') {
     finalOptions.enableNative = false;
     finalOptions.enableNativeNagger = false;
   } else {
     // keep the original value if user defined it.
     finalOptions.enableNativeNagger ??= true;
     finalOptions.enableNative ??= true;
   }
+  //  const capacitorHub = new Hub(undefined, new CapacitorScope());
+  //  makeMain(capacitorHub);
+  const defaultIntegrations: false | Integration[] =
+    passedOptions.defaultIntegrations === undefined
+      ? getDefaultIntegrations(passedOptions)
+      : passedOptions.defaultIntegrations;
 
-  const capacitorHub = new Hub(undefined, new CapacitorScope());
-  makeMain(capacitorHub);
+  finalOptions.integrations = getIntegrationsToSetup({
+    integrations: safeFactory(passedOptions.integrations, {
+      loggerMessage: 'The integrations threw an error',
+    }),
+    defaultIntegrations,
+  });
 
-  finalOptions.defaultIntegrations = [
-    ...defaultIntegrations,
-    createCapacitorRewriteFrames(),
-    new Release(),
-    new SdkInfo(),
-    new EventOrigin(),
-  ];
+  if (
+    finalOptions.enableNative &&
+    !passedOptions.transport &&
+    NATIVE.platform !== 'web'
+  ) {
+    finalOptions.transport = passedOptions.transport || makeNativeTransport;
 
+    finalOptions.transportOptions = {
+      ...(passedOptions.transportOptions ?? {}),
+      bufferSize: DEFAULT_BUFFER_SIZE,
+    };
+  }
+  useEncodePolyfill();
```

Comment:
Since capacitor uses WebView to render the application, on modern operation systems the TextEncoder should be available. 

@lucas-zimerman Can you check this, possibly we already only support systems which have TextEncoder, if not we could first check if TextEncoder is available and only use the JS impl as fallback.

Reply:
I like the idea of check if TextEncoder is available and use the JS imp as a fallback, I will implement it.